### PR TITLE
DAOS-13292 build: Don't need UCX libraries …

### PR DESCRIPTION
**Pull request by brianjmurrell**
At build time any more, as of e01970d.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by daltonbohning — APPROVED** [review]

File: site_scons/prereq_tools/base.py [hunk 1]
```diff
@@ -523,8 +523,7 @@ def __init__(self, env, opts):
 
     def run_build(self, opts):
         """Build and dependencies"""
-        # argobots is not really needed by client but it's difficult to separate
-        common_reqs = ['ucx', 'ofi', 'hwloc', 'mercury', 'boost', 'uuid', 'crypto', 'protobufc',
+        common_reqs = ['ofi', 'hwloc', 'mercury', 'boost', 'uuid', 'crypto', 'protobufc',
```
Comment (jolivier23) [inline site_scons/prereq_tools/base.py:526]: shouldn't we also remove ucx references in utils/build.config ?

Comment (jolivier23) [reply site_scons/prereq_tools/base.py:526]: may as well also remove the recipe in site_scons/components/__init__.py too?

Comment (brianjmurrell) [reply site_scons/prereq_tools/base.py:526]: Yes, I suppose so.  Will do.

**Review by daltonbohning — CHANGES_REQUESTED** [review]
We can't remove the UCX build without removing mercury's dependency on UCX. The source build fails
```
MissingDefinition: No definition for ucx:
  File "/home/dbohning/daos/SConstruct", line 512:
    scons()
  File "/home/dbohning/daos/SConstruct", line 451:
    prereqs.run_build(opts)
  File "/home/dbohning/daos/site_scons/prereq_tools/base.py", line 558:
    self.__env.Clone().require(comp)
  File "/usr/lib/python3.6/site-packages/SCons/Environment.py", line 219:
    return self.method(*nargs, **kwargs)
  File "/home/dbohning/daos/site_scons/prereq_tools/base.py", line 826:
    raise error
  File "/home/dbohning/daos/site_scons/prereq_tools/base.py", line 816:
    if comp_def.build(env, needed_libs):
  File "/home/dbohning/daos/site_scons/prereq_tools/base.py", line 1463:
    changes = self.prereqs.require(envcopy, *self.requires, needed_libs=None)
  File "/home/dbohning/daos/site_scons/prereq_tools/base.py", line 793:
    raise MissingDefinition(comp)
```

And if removing UCX 100% from the source build, I think @jgmoore-or  would be affected

**Comment by brianjmurrell** [comment]
> We can't remove the UCX build without removing mercury's dependency on UCX. The source build fails

It only fails as of 17cf0b56cc0345a992ea35787de931943364dd2a (i.e. applying @jolivier23's suggestion).  Prior to that it works.

**Comment by brianjmurrell** [comment]
I have reverted the changes that remove UCX support entirely due to the build failures it causes which correlate with the comments @daltonbohning had made about the same.  Please re-review.

**Review by daltonbohning — APPROVED** [review]

**Comment by brianjmurrell** [comment]
3 weeks later, still looking for reviews on this PR from anyone other than @daltonbohning who has already :heavy_check_mark: 'd it.

**Review by daltonbohning — DISMISSED** [review]

**Comment by brianjmurrell** [comment]
Once again, requesting reviews on this.  If you are receiving this message and you are not @daltonbohning, please review.  Thanks in advance.

**Review by jgmoore-or — APPROVED** [review]

**Comment by brianjmurrell** [comment]
> Shouldn't this PR run the Functional Hardware Medium UCX Provider stage?

I've merged master again and pushed with `Skip-func-hw-test-medium-ucx-provider: false`.

**Review by daltonbohning — APPROVED** [review]

**Comment by brianjmurrell** [comment]
Can/should this be backported to 2.6?

**Comment by daltonbohning** [comment]
> Can/should this be backported to 2.6?

Maybe add the request_for_2.6.2 label to the ticket and let them decide?

**Pull request merged**